src/client/Client.java

```diff
@@ -148,23 +157,20 @@ public InQueue(ObjectInputStream read) throws IOException {
 
         @Override
         public void run() {
-            while (!quit) {
-                try {
+            try {
+                while (!quit) {
                     Message message = (Message) read.readObject();
-                    in.add(message);
-
-                    if(gui == null) {
+                    if (message != null) {
+                        in.add(message);
                         System.out.println(message.getContent());
-                    } else{
-                        gui.updateChatArea(message);
-                    }
-
-                } catch (IOException | ClassNotFoundException e) {
-                    if (!quit) {
-                        System.out.println("Error in InQueue: " + e.getMessage());
-                        quit = true;
                     }
                 }
+            } catch (IOException | ClassNotFoundException e) {
+                if (!quit) {
+                    System.out.println("Error in InQueue: " + e.getMessage());
+                }
+            } finally {
+                quit(); // Ensure cleanup when the thread exits
```

Comment:
I think this will result in an unnecessary duplicate call to quit(). The InQueue and OutQueue threads are supposed to be managed by the client, so when the client closes, it is responsible for terminating the two queue threads via a call to quit. This will set the quit flag to true, causing the threads to terminate.